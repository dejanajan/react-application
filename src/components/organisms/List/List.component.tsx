import React from "react";
import * as ListStyle from "./List.styles";

interface ListType {
  list: any[];
}

export function List(props: ListType) {
  const { list } = props;

  return (
    <ListStyle.List>      
      {
        list.map((item: { id: number; name: string; html_url: string}) => (
          <li key={item.id}>
            <a href={item.html_url}>{item.name}</a>
          </li>
        ))
      }
    </ListStyle.List>
  );
}


export default List