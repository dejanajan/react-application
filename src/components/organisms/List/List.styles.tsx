import styled from "styled-components";

type Custom = {
    color?: string
    hover?: string,
}

export const List = styled.ul<Custom>`
  width: 100%;
  max-width: 400px;
  li {
      list-style-type: square;
    a {
        font-size: 16px;
        color: ${props => props.color || "tomato"};
        &:hover {
            font-weight: 600;
            color: ${props => props.hover || "blue"};
        }
    }
  }
`;