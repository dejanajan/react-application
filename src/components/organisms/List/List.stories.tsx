import React from "react";

import { storiesOf } from "@storybook/react";
import List from './List.component';


const numbers = [{id: 255900773, name: "Test"}, 
                 {id: 255900756, name: "Angular"}, 
                 {id: 255900759, name: "React"}
                ];

storiesOf("List", module)
    .add("default", () => (
    <List list={numbers} />
  ));
  
