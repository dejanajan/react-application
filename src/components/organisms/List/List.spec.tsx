import React from "react";
import { Source } from "../../../models/repos";
import List from "./List.component";
import renderer from "react-test-renderer";

const numbers: Source[] | any[] = [{id: 255900773, name: "Test"}, 
                 {id: 255900756, name: "Angular"}, 
                 {id: 255900759, name: "React"}
                ];

test("Render List", () => {

  const list = renderer.create(<List list={numbers} />);
  const tree = list.toJSON();

  expect(tree).toMatchSnapshot();
});
