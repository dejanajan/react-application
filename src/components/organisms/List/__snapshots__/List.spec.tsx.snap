// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render List 1`] = `
<ul
  className="sc-AxjAm bVsHij"
>
  <li>
    <a>
      Test
    </a>
  </li>
  <li>
    <a>
      Angular
    </a>
  </li>
  <li>
    <a>
      React
    </a>
  </li>
</ul>
`;
